Modules/Server/Sources/PocketCastsServer/Public/ServerPodcastManager+Update.swift

```diff
@@ -155,6 +166,10 @@ extension ServerPodcastManager {
             // for subscribed podcasts remove any non-interacted with episodes that aren't in the server JSON
             cleanupDeletedEpisodes(podcast: podcast, serverEpisodes: episodesJson)
         }
+
+        if missingEpisodesThatIsNotTheLastOne {
+            RefreshManager.shared.refresh(podcast: podcast)
```

Comment:
This looks like it will download all the podcast episodes sync information. Android uses the following call to download the single episode sync data. It's handy for spots like the Up Next where you might not be subscribed to the podcast but require the user sync information. 


```
    @GET("/mobile/podcast/findbyepisode/{podcastUuid}/{episodeUuid}")
    fun getPodcastAndEpisodeSingle(@Path("podcastUuid") podcastUuid: String, @Path("episodeUuid") episodeUuid: String): Single<PodcastResponse>
```

Reply:
@geekygecko we don't even need to make an additional request if we don't want to, given that during this refresh we already have the episode data. This wasn't done for a few reasons:

1. There was a comment explicitly stating that new episodes for subscribed podcasts are skipped by this logic because they might need to be auto-downloaded and auto-added to Up Next. A "podcast refresh" handles that, this logic doesn't;
2. If we just add the episode — without any other call — we have the possibility of adding an episode without any status. So, let's say a user started or even played this podcast on another platform, we won't update it accordingly.

This logic should be triggered if, and **only if**:

1. A podcast has episodes "in the middle" that weren't added before;
2. This should affect pretty much ONLY podcasts that change their publish dates leading the app to "ignore" some episodes

In terms of load on the server, I don't expect much change because is a very specific case.

However, we can take a different approach:

1. If the episode is "in the middle" we just add it right away. No additional calls, no sync.

The downside is that if the user has any special setting for this podcast (like Auto Download and/or Add to Up Next), we won't do it. Also, if they interacted with this episode on another platform we won't sync that either (like never).

It's still better as the current state (in which the episode NEVER appears) but it sounds like a half-fix. 😬

Let me know your thoughts!

Reply:
@geekygecko just to keep you in the loop: we're merging this but it's behind a remote feature flag. The logic was changed to refresh only the necessary to add the missing episodes. Plus it's only triggered in a very specific path. If we run into any issues we can just disable the flag.